pkg/stub/handler.go

```diff
@@ -62,16 +62,48 @@ func (h *Handler) EnsureDefaultClusterIngress() error {
 		return err
 	}
 
-	err = sdk.Create(ci)
-	if err == nil || errors.IsAlreadyExists(err) {
-		if err == nil {
-			logrus.Infof("created default cluster ingress %s/%s", ci.Namespace, ci.Name)
+	changed, notFound, err := isClusterIngressChanged(ci)
+	if err != nil {
+		return err
+	}
+	if changed {
+		err = sdk.Update(ci)
+		if err != nil {
+			return fmt.Errorf("updating default cluster ingress %s/%s: %v", ci.Namespace, ci.Name, err)
 		}
-
-		return nil
+		logrus.Infof("updated default cluster ingress %s/%s", ci.Namespace, ci.Name)
+	} else if notFound {
+		err = sdk.Create(ci)
+		if err != nil {
+			return fmt.Errorf("creating default cluster ingress %s/%s: %v", ci.Namespace, ci.Name, err)
+		}
+		logrus.Infof("created default cluster ingress %s/%s", ci.Namespace, ci.Name)
 	}
+	return nil
+}
 
-	return fmt.Errorf("creating default cluster ingress %s/%s: %v", ci.Namespace, ci.Name, err)
+func isClusterIngressChanged(ci *ingressv1alpha1.ClusterIngress) (bool, bool, error) {
+	oldci := &ingressv1alpha1.ClusterIngress{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       ci.Kind,
+			APIVersion: ci.APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ci.Name,
+			Namespace: ci.Namespace,
+		},
+	}
+	err := sdk.Get(oldci)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return false, false, fmt.Errorf("failed to fetch existing default cluster ingress %s/%s, %v", ci.Namespace, ci.Name, err)
+		}
+		return false, true, nil
+	}
+	if oldci.Spec.IngressDomain != ci.Spec.IngressDomain {
```

Comment:
I think you need to deference the pointers to do what you mean here: https://play.golang.org/p/N1-AT4yXgH-

Reply:
yes, thanks